src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,77 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { ChangeEvent } from 'react';
+import { FilterType } from '../../types/FilterTypes';
+import './TodoFilter.scss';
+
+type Props = {
+  query: string;
+  onQueryChange: (query: string) => void;
+  selectedType: string;
+  onSelectedType: (filter: FilterType) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  onQueryChange,
+  selectedType,
+  onSelectedType,
+}) => {
+  const handleQuery = (event: ChangeEvent<HTMLInputElement>) => {
+    onQueryChange(event.target.value);
+  };
+
+  const resetQuery = () => {
+    onQueryChange('');
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            className="select-type"
+            data-cy="statusSelect"
+            value={selectedType}
+            onChange={event => {
+              const target = event.target.value;
+
+              return (
+                onSelectedType(target as FilterType)
+              );
```

Comment:
It's better to fix the type as soon as you get it before usage because you can use it in a few places and you don't want to write `as FilterType` in a few places. With this change, your variable `target` already has the correct type.
```suggestion
              const target = event.target.value as FilterType;

              return (
                onSelectedType(target)
              );
```